tests/components/sensor/test_islamic_prayer_times.py

```diff
@@ -0,0 +1,160 @@
+        mock_pt_calc.PrayerTimesCalculator.return_value.fetch_prayer_times \
+            .return_value = PRAYER_TIMES
+
+        pt_data = IslamicPrayerTimesData(latitude=LATITUDE,
+                                         longitude=LONGITUDE,
+                                         calc_method=CALC_METHOD)
+
+        assert pt_data.get_new_prayer_times() == PRAYER_TIMES
+        assert pt_data.get_prayer_times_info() == PRAYER_TIMES
+
+
+async def test_islamic_prayer_times_sensor_initial_state(hass):
+    """Test Islamic prayer times sensor initial state."""
+    with MockDependency('prayer_times_calculator') as mock_pt_calc:
+        mock_pt_calc.PrayerTimesCalculator.return_value.fetch_prayer_times \
+            .return_value = PRAYER_TIMES
+
+        config = {
+            'sensor': {
+                'platform': 'islamic_prayer_times',
+                'sensors': ['maghrib']
+            }
+        }
+
+        assert await async_setup_component(hass, 'sensor', config)
+
+        entity_id = 'sensor.islamic_prayer_time_maghrib'
+        pt_dt = get_prayer_time_as_dt(PRAYER_TIMES['Maghrib'])
+        entity = hass.states.get(entity_id)
+        assert entity.state == pt_dt.isoformat()
+        assert entity.name == 'Maghrib'
+
+
+async def test_islamic_prayer_times_sensor_update(hass):
+    """Test Islamic prayer times sensor update."""
+    with MockDependency('prayer_times_calculator') as mock_pt_calc:
+        mock_pt_calc.PrayerTimesCalculator.return_value.fetch_prayer_times \
+            .return_value = PRAYER_TIMES
+
+        config = {
+            'sensor': {
+                'platform': 'islamic_prayer_times',
+                'sensors': ['maghrib']
+            }
+        }
+
+        assert await async_setup_component(hass, 'sensor', config)
+
+        entity_id = 'sensor.islamic_prayer_time_maghrib'
+        pt_dt = get_prayer_time_as_dt(PRAYER_TIMES['Maghrib'])
+        entity = hass.states.get(entity_id)
+        assert entity.state == pt_dt.isoformat()
+
+        midnight = PRAYER_TIMES['Midnight']
+
+        new_prayer_times = {"Fajr": "06:10",
+                            "Sunrise": "07:25",
+                            "Dhuhr": "12:30",
+                            "Asr": "15:32",
+                            "Maghrib": "17:45",
+                            "Isha": "18:53",
+                            "Midnight": "00:45"}
+
+        print("New Prayer Times: {}".format(str(new_prayer_times)))
+
+        now = dt_util.as_local(dt_util.now())
+        today = now.date()
+
+        midnight_dt_str = '{}::{}'.format(str(today), midnight)
+        midnight_dt = datetime.strptime(midnight_dt_str, '%Y-%m-%d::%H:%M')
+        future = midnight_dt + timedelta(days=1, minutes=1)
+
+        with patch(('homeassistant.components.sensor.islamic_prayer_times.'
+                    'dt_util.utcnow'), return_value=future):
+
+            with patch(('homeassistant.components.sensor.islamic_prayer_times.'
```

Comment:
A single `with` statement can take multiple comma separated patches.

Reply:
@MartinHjelmare doing it with a single with statement is giving me the following error:

```
E                           AttributeError: __enter__

tests/components/sensor/test_islamic_prayer_times.py:151: AttributeError
```

here's what it looks like with a single with:

```python
        with (patch(
                "homeassistant.components.sensor.islamic_prayer_times"
                ".dt_util.utcnow", return_value=future),
                    patch(
                        "homeassistant.components.sensor.islamic_prayer_time"
                        ".IslamicPrayerTimesData.get_prayer_times_info",
                        return_value=new_prayer_times)):
```

Reply:
Maybe the outer parenthesis is giving you trouble? If it's hard to find the correct indentation without extra parenthesis, try saving the patches in two variables before calling `with` to make that line shorter. But it's not required.

Reply:
Removing the outer parenthesis gives me an `Illegal target for variable annotation error`: 
<img width="572" alt="image" src="https://user-images.githubusercontent.com/1053584/50235896-03c19f80-0387-11e9-831d-87276d5d60f0.png">

I'll try the multiple variable route but if that doesn't work, is it okay to keep the dual `with` statements like i had it before?

Reply:
alright figured it out.  pycharm's line-ending were causing a lot of issues.